### ISSUE TITLE
Unable to write on the BMS

### ISSUE THREAD

**Issue by atom-2jong** [NeariX67/SmartBMSUtility#19]
![9CACD2FA-C31F-4958-9B90-C43DECE3C878](https://user-images.githubusercontent.com/81765688/163786110-2fb096d8-8f9f-4880-9b3e-199011a6bb52.jpeg)
![A8315A8E-4794-4E3F-AAE8-322DA7FA9186](https://user-images.githubusercontent.com/81765688/163786124-39af492c-3a1d-43a6-a5c6-ff77d4c66c54.jpeg)
I can't read/write on the BMS, the button is grayed out. I can't activate charge/discharge either. However, I can do it via the XiaoxiangBMS app.

**Comment by katter** [comment]
Seeing the same behaviour on a JBD-SP04S028-L4S-60A-B. And config read/write works with XiaoxiangBMS, switching Charge/ Discharge works with neither one.

**Comment by NeariX67** [comment]
Are you using a DIY battery? I haven't seen BMS' where my protection kicks in that are not from LionTron.
I'm gonna order that particular bms and test it out

**Comment by katter** [comment]
Oh wait, it's a "Wipuvo" battery pack. Is there any way, I can do something to provide you with helpful data?

**Comment by atom-2jong** [comment]
Yes, it's a DIY battery. The first time I launched the Smart BMS utility app, it worked, I could change the settings and save them as well as activate charging/discharge. But then I changed the BMS settings with the XiaoxiangBMS app and since then it no longer works with Smart BMS utility.

**Comment by NeariX67** [comment]
What my guess is for both of your cases is that you used the xiaoxiang app to set a hardware password, which is not yet supported by my app. I tested the Xiaoxiang app to some extend and noticed some issues regarding the hardware pin, but he was unwilling to fix it.
https://github.com/smagicld/xiaoxiangBMS/issues/177
https://github.com/smagicld/xiaoxiangBMS/issues/176

So far - wait for version 2 of this app, currently working on it. You may test the beta of v2 with the testflight link in the description (WARNING: Missing important features so far)

**Comment by atom-2jong** [comment]
Thanks ! But the beta is complete, I can't join it…

**Comment by atom-2jong** [comment]
![Uploading B65CCEE8-F43B-47A6-AFB6-B06A070A3E62.png…]()

**Comment by atom-2jong** [comment]
![593B711B-B8D6-483A-8A91-394572361AF6](https://user-images.githubusercontent.com/81765688/165153554-d811dba9-1987-4ef1-9a4c-0450410cdff0.png)

**Comment by NeariX67** [comment]
Oh it probably expired already. I hope I can submit a new release today or in the next few days.

**Comment by NeariX67** [comment]
Just uploaded a new beta of v2. Should be available through the invitation link.

**Comment by atom-2jong** [comment]
Thanks you so much !!
